sklearn/decomposition/_dict_learning.py

```diff
@@ -487,10 +487,12 @@ def dict_learning(X, n_components, *, alpha, max_iter=100, tol=1e-8,
         for more details.
 
     dict_init : ndarray of shape (n_components, n_features), default=None
-        Initial value for the dictionary for warm restart scenarios.
+        Initial value for the dictionary for warm restart scenarios. Only used
+        if `code_init` is not None.
```

Comment:
I think this is slightly clearer:

```suggestion
        Initial value for the dictionary for warm restart scenarios. Only used
        if `code_init` and `dict_init` are not None.
```

If `dict_init` is None and `code_init` is not None, I do not think we "use" `None`. `dict_learning` will still call `svd` to compute the `code` and `dictionary`.

Reply:
I agree